lib/msal-common/src/utils/RequestUtils.ts

```diff
@@ -0,0 +1,248 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { AADServerParamKeys, SSOTypes } from "./Constants";
+import { Constants, HeaderNames } from "./Constants";
+import { ScopeSet } from "../request/ScopeSet";
+import { ClientConfigurationError } from "../error/ClientConfigurationError";
+
+export class RequestUtils {
```

Comment:
This can be trimmed to a non side effect function  like getparamsfromrequest(request)
map = new map()
if request.something
    map.addSomething
if map.somethingElse
   mapp.addSomethingElse


the benefit is that this function will be idempotent. Given the same input the output will always be the same, when passing in an object to alter, its too easy to end up in hard to trace conditions.